mypy/types.py

```diff
@@ -43,6 +43,12 @@ def accept(self, visitor: 'TypeVisitor[T]') -> T:
     def __repr__(self) -> str:
         return self.accept(TypeStrVisitor())
 
+    def __hash__(self) -> int:
+        return hash(type(self))
+
+    def __eq__(self, other: object) -> bool:
```

Comment:
Don't implement ``__eq__`` and ``__hash__`` on ``Type`` or make them raise ``NotImplementedError``. (Otherwise, if one forgets to implement them on a subclass this could cause obscure bugs.)